src/clr-icons/clr-icons-element.ts

```diff
@@ -67,10 +71,19 @@ ClarityIconElement.prototype.connectedCallback = function() {
   // So we could check whether the attribute values really changed or not.
   // If not, we don't need to execute the same codes again.
 
-  // We want to hide the custom element from screen readers but allow the svg/img content to still be read inline
-  // Adding role=none allows the screen reader to skip the custom element as if it were a div or span.
-  // https://www.scottohara.me/blog/2018/05/05/hidden-vs-none.html
-  this.setAttribute('role', 'none');
+  if (this.hasAttribute('role')) {
```

Comment:
Wait, why do we need this? If the user places this button role on the icon, that should be it. Why do we set it again on the icon again here?

Reply:
adding the role of 'none' hides this from the screen readers.
Even with role='button on the clr-icon element screen readers will ignore this.

Reply:
You have clrTooltipTrigger that will add a role value to the icon here as you defined in the host. I don't understand what you are trying to do here. If it has to have the 'button' role, the directive will place that, no need to change anything inside the clr-icon implementation.  

When or why should it have the role 'none'?

Reply:
This overrides anything that the directive does. Adding role to a clr-icon will always get whatever this code says. In my investigation I found that this is the reason screen readers were ignoring it.

Do you have suggestions for informing screen readers with a role of button and keeping the role of none here?

Reply:
Role of none was added here to address other a11y issues: https://github.com/vmware/clarity/pull/3448/commits/4d3d1a2aa5707b426d998c3de780986698bffa32

Reply:
That's by default, it has 'none' role. Here, the tooltip trigger will override that `none` role with `button`. That's it. You don't have to do anything in `clr-icon` element.

Reply:
No, I tried. It always gets a role of none no matter what the directive code does.

Reply:
Whatever role the directive assigns to the icon, it will stay on `clr-icon` https://stackblitz.com/edit/clarity-light-theme-v2-4y5ayd

Reply:
Resolved this by switching the directive to attribute binding instead of setting the property directly. That eliminates the need for this change. Will remove.